File: Go_Assignment/rSort.go
package main

func numsSort(nums []int) []int {
	result := make([]int, len(nums))
	copy(result, nums)
	for i := range result {
		if i > 0 {
			for preI := i; preI > 0 && result[preI] < result[preI-1]; preI-- {
				result[preI], result[preI-1] = result[preI-1], result[preI]
			}
		}
	}
	return result
}

func strsSort(strs []string) []string {
	result := make([]string, len(strs))
	copy(result, strs)
	for i := range result {
		if i > 0 {
			for preI := i; preI > 0 && result[preI] < result[preI-1]; preI-- {
				result[preI], result[preI-1] = result[preI-1], result[preI]
			}
		}
	}
	return result
}

func rationalSort(rs []Rationalizer) []Rationalizer {
	result := make([]Rationalizer, len(rs))
	copy(result, rs)
	for i := range rs {
		if i > 0 {
			for preI := i; preI > 0 && result[preI].LessThan(result[preI-1]); preI-- {
				result[preI], result[preI-1] = result[preI-1], result[preI]
			}
		}
	}
	return result
}

File: Ruby_Assignment/rSort.rb
def numsSort(nums)
    result = nums.dup
    result.each_with_index do |n,i|
        if i > 0
            preI = i
            while preI > 0 && result[preI] < result[preI-1] do
                result[preI], result[preI-1] = result[preI-1], result[preI]
                preI = preI - 1
            end
        end
    end
    return result
end


def strsSort(strs)
    result = strs.dup
    result.each_with_index do |s,i|
        if i > 0
            preI = i
            while preI > 0 && result[preI] < result[preI-1] do
                result[preI], result[preI-1] = result[preI-1], result[preI]
                preI = preI - 1
            end
        end
    end
    return result
end

def rationalsSort(rs)
    result = rs.dup
    result.each_with_index do |r,i|
        if i > 0
            preI = i
            while preI > 0 && result[preI] < result[preI-1] do
                result[preI], result[preI-1] = result[preI-1], result[preI]
                preI = preI - 1
            end
        end
    end
    return result
end